netplan/cli/commands/ip.py

```diff
@@ -90,21 +90,19 @@ def lease_method_ifindex():
                     logging.debug('Cannot read file %s: %s', ifindex_f, str(e))
                     raise
 
-            def lease_method_nm_connection():  # pragma: nocover (covered in autopkgtest)
+            def lease_method_nm_connection():
                 # FIXME: handle older versions of NM where 'nmcli dev show' doesn't exist
                 try:
-                    nmcli_dev_out = subprocess.Popen(['nmcli', 'dev', 'show', self.interface],
-                                                     env={'LC_ALL': 'C'},
-                                                     stdout=subprocess.PIPE)
-                    for line in nmcli_dev_out.stdout:
-                        line = line.decode('utf-8')
+                    nmcli_dev_out = subprocess.check_output(['nmcli', 'dev', 'show', self.interface],
+                                                            env=dict(LC_ALL='C', PATH=os.environ.get('PATH')),
+                                                            universal_newlines=True)
+                    for line in nmcli_dev_out.splitlines():
                         if 'GENERAL.CONNECTION' in line:
                             conn_id = line.split(':')[1].rstrip().strip()
-                            nmcli_con_out = subprocess.Popen(['nmcli', 'con', 'show', 'id', conn_id],
-                                                             env={'LC_ALL': 'C'},
-                                                             stdout=subprocess.PIPE)
-                            for line in nmcli_con_out.stdout:
-                                line = line.decode('utf-8')
+                            nmcli_con_out = subprocess.check_output(['nmcli', 'con', 'show', 'id', conn_id],
+                                                                    env=dict(LC_ALL='C', PATH=os.environ.get('PATH')),
```

Comment:
`subprocess` can't deal with an environment having a None value, so I would suggest:
```diff
-os.environ.get('PATH')
+os.environ['PATH']
```
Unless you wanted to use `os.environ.get('PATH', os.defpath)` here ?
For sure it will also fail if PATH is not present in the environment, but the error should be less obscure, I think.

Reply:
Thank you, good spot! I opted to using `os.environ.get('PATH', os.defpath)` as this should give the least obscure errors (i.e. no errors about the environment, but about `nmcli` if it does not exist).